PlayChat/AppDelegate.swift

```diff
@@ -14,208 +14,216 @@
  **/
 
 import Firebase
+import GoogleSignIn
 import UIKit
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate, GIDSignInDelegate,
   UITableViewDelegate,
 UITextFieldDelegate {
-  let IBX : String = "inbox"
-  let CHS : String = "channels"
-  let REQLOG : String = "requestLogger"
-  var maxMessages: UInt = 20
-  
+  let IBX: String = "inbox"
+  let CHS: String = "channels"
+  let REQLOG: String = "requestLogger"
+  var chanArray: [String] = []
+  var maxMessages: UInt?
+
   var window: UIWindow?
   var storyboard: UIStoryboard?
   var navigationController: UINavigationController?
-  var tabBarController: UITabBarController!
-  var msgViewController : MessageViewController?
-  
+  var tabBarController: UITabBarController?
+  var msgViewController: MessageViewController?
+
   var user: GIDGoogleUser!
   var inbox: String?
-  var ref: FIRDatabaseReference!
+  var ref: DatabaseReference!
   var msgs: [Message] = []
-  var channelViewDict: [String : UITableView] = [:]
+  var channelViewDict: [String: UITableView] = [: ]
   var fbLog: FirebaseLogger?
-  
-  func application(application: UIApplication, didFinishLaunchingWithOptions
-    launchOptions: [NSObject: AnyObject]?) -> Bool {
-    var configureError: NSError?
-    GGLContext.sharedInstance().configureWithError(&configureError)
-    assert(configureError == nil, "Config error: \(configureError)")
-    
-    let path = NSBundle.mainBundle().pathForResource("Info", ofType: "plist")!
-    let dict = NSDictionary(contentsOfFile: path) as! [String: AnyObject]
-    let channels = dict["Channels"] as! String
-    let chanArray = channels.componentsSeparatedByString(",")
-    maxMessages = dict["MaxMessages"] as! UInt
+
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions
+    launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+
+    if let path = Bundle.main.path(forResource: "Info", ofType: "plist"),
+       let dictionary = NSDictionary(contentsOfFile: path) as? [String: AnyObject] {
+      chanArray = dictionary["Channels"]!.components(separatedBy: ",")
+      maxMessages = dictionary["MaxMessages"]! as? UInt
+    } else {
+      print("Couldn't read 'Info.plist' file")
+    }
 
     storyboard = UIStoryboard(name: "Main", bundle: nil)
     navigationController = storyboard!.instantiateInitialViewController()
       as? UINavigationController
+
     tabBarController = self.storyboard!
-      .instantiateViewControllerWithIdentifier("TabBarController")
-      as! UITabBarController
-    msgViewController = MessageViewController(maxMessages: maxMessages)
+        .instantiateViewController(withIdentifier: "TabBarController")
+        as? UITabBarController
+    msgViewController = MessageViewController(maxMessages: maxMessages!)
```

Comment:
Code-practices-wise, it's better to have a default value and use it here if `maxMessages` is nil.

Reply:
Sounds good. I just need some clarification.
Do you mean to have a default parameter value on the init method of `MessageViewController`, as follows:
```
init(maxMessages: UInt = 20) {
  ...
}
```
OR

do you mean to have the default value on line 28 on this file, as follows:
```
var maxMessages: UInt = 20
```

I went with the 2nd option, but I'll be happy to update if that's not what you meant.

Reply:
This LGTM